source/Halibut/ServiceModel/HalibutProxy.cs

```diff
@@ -144,12 +146,23 @@ RequestMessage CreateRequest(MethodInfo targetMethod, object[] args)
             };
             return request;
         }
-
-        ResponseMessage DispatchRequest(RequestMessage requestMessage)
+        
+#endif
+        ResponseMessage DispatchRequest(RequestMessage requestMessage, CancellationToken connectCancellationToken)
         {
-            return messageRouter(requestMessage, cancellationToken);
+            return messageRouter(requestMessage, connectCancellationToken);
         }
-#endif
+        CancellationToken ConnectingCancellationToken(HalibutProxyRequestOptions halibutProxyRequestOptions)
+        {
+            if (halibutProxyRequestOptions == null || halibutProxyRequestOptions.ConnectCancellationToken == null)
+            {
+                return globalCancellationToken;
+            }
+
+            CancellationToken ct = (CancellationToken) halibutProxyRequestOptions.ConnectCancellationToken;
+            return CancellationTokenSource.CreateLinkedTokenSource(globalCancellationToken, ct).Token;
```

Comment:
Is this making it even more confusing e.g. we can have 2 cancellation tokens now both cancelling connects? One for the whole proxy and one for just a request.

Thoughts on changing this so you can only have one or the other e.g. if the global one is provided (anything other than CT.None) blow up when a request one is provided.

Reply:
Solved by only allowing one token to be set by the user.